test/constrained-generics/special-method-warnings/user-hash-warning-generic-generic-impl.good

```diff
@@ -0,0 +1,3 @@
+user-hash-warning-generic-generic-impl.chpl:3: warning: the type 'R' defines methods that previously had special meaning. These will soon require 'R' to implement the 'hashable' interface to continue to be treated specially.
+user-hash-warning-generic-generic-impl.chpl:7: note: formerly-special function 'hash' defined here
```

Comment:
Why does this test trigger the warning but the one right above it doesn't?

Reply:
For some reason, concrete secondary methods __that aren't invoked__ have an `unknown` receiver type (?), which makes it hard to figure out which type they need to implement `hashable`. If you call `hash`, the receiver type is determined; if the receiver type is generic, it is never `unknown`. The `.future` added by this PR notes this behavior; I'll open an issue for the missing warning as of fixing your comment above.

Reply:
https://github.com/chapel-lang/chapel/issues/23189